<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Getting started with scRNA-seq • pluto</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><link href="../extra.css" rel="stylesheet">
<meta property="og:title" content="Getting started with scRNA-seq">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">pluto</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../articles/authentication.html">Auth</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-get-started" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Get started</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-get-started">
<li><a class="dropdown-item" href="../articles/get_started.html">Quick start guide</a></li>
    <li><a class="dropdown-item" href="../articles/pluto_models.html">Intro to the Pluto data models</a></li>
  </ul>
</li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-vignettes" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Vignettes</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-vignettes">
<li><h6 class="dropdown-header" data-toc-skip>Common recipes by assay type</h6></li>
    <li><a class="dropdown-item" href="../articles/rnaseq_recipes.html">RNA-seq</a></li>
    <li><a class="dropdown-item" href="../articles/scrnaseq_recipes.html">Single cell RNA-seq</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Advanced analyses</h6></li>
    <li><a class="dropdown-item" href="../articles/data_integration.html">Introduction to multi-experiment integration</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item"><a class="external-link nav-link" href="https://help.pluto.bio"><span class="fa fa-external-link"></span> Help Center</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/pluto-biosciences/pluto-sdk-R/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Getting started with scRNA-seq</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/pluto-biosciences/pluto-sdk-R/blob/main/vignettes/scrnaseq_recipes.Rmd" class="external-link"><code>vignettes/scrnaseq_recipes.Rmd</code></a></small>
      <div class="d-none name"><code>scrnaseq_recipes.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="quick-start-guide">Quick start guide<a class="anchor" aria-label="anchor" href="#quick-start-guide"></a>
</h2>
<p>See <code><a href="../articles/authentication.html">vignette("authentication")</a></code> for instructions on
creating your personal API token for accessing data. To begin using
<code>pluto</code>, load the package and log in with your personal API
token:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://pluto.bio" class="external-link">pluto</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/pluto_login.html">pluto_login</a></span><span class="op">(</span><span class="st">"&lt;YOUR API TOKEN&gt;"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="accessing-seurat-objects-from-pluto">Accessing <code>Seurat</code> objects from Pluto<a class="anchor" aria-label="anchor" href="#accessing-seurat-objects-from-pluto"></a>
</h2>
<p>Single cell RNA-seq in Pluto creates two <code>Seurat</code> objects:
a <strong>raw</strong> <code>Seurat</code> object, which is created at
the beginning of the single cell RNA-seq preprocessing workflow and has
not been filtered, normalized, integrated, clustered, etc.; and a
<strong>final</strong> <code>Seurat</code> object, which is the final,
processed <code>Seurat</code> object that results from completing and
accepting a preprocess workflow. Both <code>Seurat</code> objects are
accessible for external scRNA-seq analysis via the <code>pluto</code>
package.</p>
<div class="section level3">
<h3 id="download-seurat-object">Download Seurat object<a class="anchor" aria-label="anchor" href="#download-seurat-object"></a>
</h3>
<p>To download a <code>Seurat</code> object, provide the experiment ID,
<code>Seurat</code> type (<code>"raw"</code> or <code>"final"</code>) to
the <code><a href="../reference/pluto_download_seurat_object.html">pluto_download_seurat_object()</a></code> function (you can also
provide an optional desired destination file path by setting
<code>dest_filename</code>):</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">experiment_id</span> <span class="op">&lt;-</span> <span class="st">"PLX037839"</span></span>
<span><span class="va">seurat_type</span> <span class="op">&lt;-</span> <span class="st">"raw"</span></span>
<span><span class="fu"><a href="../reference/pluto_download_seurat_object.html">pluto_download_seurat_object</a></span><span class="op">(</span><span class="va">experiment_id</span>, <span class="va">seurat_type</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="read-seurat-object">Read <code>Seurat</code> object<a class="anchor" aria-label="anchor" href="#read-seurat-object"></a>
</h3>
<p>To read a <code>Seurat</code> object directly into your R script,
provide the experiment ID and Seurat type (<code>"raw"</code> or
<code>"final"</code>) to the <code><a href="../reference/pluto_read_seurat_object.html">pluto_read_seurat_object()</a></code>
function:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">experiment_id</span> <span class="op">&lt;-</span> <span class="st">"PLX037839"</span></span>
<span><span class="va">seurat_type</span> <span class="op">&lt;-</span> <span class="st">"raw"</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pluto_read_seurat_object.html">pluto_read_seurat_object</a></span><span class="op">(</span><span class="va">experiment_id</span>, <span class="va">seurat_type</span><span class="op">)</span></span></code></pre></div>
<p>It is important to note that the
<code><a href="../reference/pluto_read_seurat_object.html">pluto_read_seurat_object()</a></code> function returns different
objects depending on the <code>Seurat</code> type specified. If the
<code>Seurat</code> type is specified as <code>"raw"</code>, a Seurat
object will be returned. If the <code>Seurat</code> type is specified as
<code>"final"</code>, a list is returned containing the final
<code>Seurat</code> object, <code>obj</code>, with custom cluster
annotations added to the <code>Seurat</code> <code>meta.data</code> and
custom color palettes corresponding to the cluster annotation sets
in-app, <code>colors</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">experiment_id</span> <span class="op">&lt;-</span> <span class="st">"PLX037839"</span></span>
<span><span class="va">seurat_type</span> <span class="op">&lt;-</span> <span class="st">"final"</span></span>
<span><span class="va">final_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pluto_read_seurat_object.html">pluto_read_seurat_object</a></span><span class="op">(</span><span class="va">experiment_id</span>, <span class="va">seurat_type</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># To access the Seurat object</span></span>
<span><span class="va">so</span> <span class="op">&lt;-</span> <span class="va">final_obj</span><span class="op">$</span><span class="va">obj</span></span>
<span></span>
<span><span class="co"># To access the custom color palettes</span></span>
<span><span class="va">custom_palettes</span> <span class="op">&lt;-</span> <span class="va">final_obj</span><span class="op">$</span><span class="va">colors</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="the-seurat-object">The <code>Seurat</code> object<a class="anchor" aria-label="anchor" href="#the-seurat-object"></a>
</h2>
<p>The <code>Seurat</code> object, <a href="https://satijalab.org/seurat/" class="external-link">developed by the Satija lab</a>, is
a representation of single-cell expression data for R. The object serves
as a container that contains both data (like the count matrix) and
analysis (like PCA, or clustering results) for a single cell dataset. At
the top level, the <code>Seurat</code> object serves as a collection of
<code>Assay</code> and <code>DimReduc</code> objects, representing
expression data and dimensionality reductions of the expression data,
respectively. <code>Seurat</code> objects also store additional
metadata, both at the cell and feature level. Click <a href="https://github.com/satijalab/seurat/wiki" class="external-link">here</a> for more
information.</p>
<p>To access your <code>Seurat</code> object assay(s):</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">assays</span></span></code></pre></div>
<p>To access your <code>Seurat</code> object reductions:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">reductions</span></span></code></pre></div>
<p>To access your <code>Seurat</code> object <code>meta.data</code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span></span>
<span></span>
<span><span class="co"># Or, to just display the first n rows of meta.data:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span></code></pre></div>
<p>The column names in your <code>meta.data</code> will correspond to
the <strong>variables</strong> and <strong>latent variables</strong> you
see in the Pluto app. You will also see additional columns that were
added during the preprocess workflow, mostly related to QC. If you are
reading in a <strong>final</strong> <code>Seurat</code> object, you will
see columns that correspond to your cluster annotation sets.</p>
<p>To get a list of available <code>meta.data</code> columns in your
<code>Seurat</code> object:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">so</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualization-using-the-seurat-package">Visualization using the <code>Seurat</code> package<a class="anchor" aria-label="anchor" href="#visualization-using-the-seurat-package"></a>
</h2>
<p>Once a <code>Seurat</code> object has been downloaded from Pluto, we
can use a variety of base plotting functions from the
<code>Seurat</code> package. We can further modify and customize these
plots using <code>ggplot2</code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://wilkelab.org/ggridges/" class="external-link">ggridges</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="dimensional-reduction-plot">Dimensional reduction plot<a class="anchor" aria-label="anchor" href="#dimensional-reduction-plot"></a>
</h3>
<p>Using the <code>Seurat</code> <code><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot()</a></code> function, make a
scatter plot in UMAP space grouping cells by custom annotation set
“Default cluster - res. 0.1” (here, we can use one of our custom color
palettes that we defined in the Pluto app):</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">so</span>, <span class="co"># the Seurat object</span></span>
<span>     reduction <span class="op">=</span> <span class="st">"umap"</span>, <span class="co"># could also be "tsne" or "pca"</span></span>
<span>     group.by <span class="op">=</span> <span class="st">"default_clusters_res_0_1"</span>, <span class="co"># custom annotation set</span></span>
<span>     cols <span class="op">=</span> <span class="va">custom_palettes</span><span class="op">$</span><span class="va">default_clusters_res_0_1</span>, <span class="co"># custom color palette</span></span>
<span>     pt.size <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="co"># point size for plotting</span></span>
<span>     <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>          title <span class="op">=</span> <span class="st">"Default clusters - res. 0.1"</span>, <span class="co"># update plot title</span></span>
<span>          x <span class="op">=</span> <span class="st">"UMAP 1"</span>, <span class="co"># update x-axis label</span></span>
<span>          y <span class="op">=</span> <span class="st">"UMAP 2"</span></span>
<span>     <span class="op">)</span> <span class="co"># update y-axis label</span></span></code></pre></div>
<p><img src="https://cdn.bfldr.com/2Q1IPX6I/at/cv9v8qtfswrqvq69gh58swqb/scrnaseq_vignette_dimplot_group_clusters_example" height="450"></p>
<p>Using the <code>Seurat</code> <code><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot()</a></code> function, make a
scatter plot in UMAP space grouping cells by timepoint:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a custom color palette to group cells by timepoint</span></span>
<span><span class="va">timepoint_cols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"#14b8a6"</span>, <span class="st">"#fcd34d"</span>, <span class="st">"#ef4444"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">timepoint_cols</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"24hpf"</span>, <span class="st">"36hpf"</span>, <span class="st">"48hpf"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">so</span>, <span class="co"># the Seurat object</span></span>
<span>     reduction <span class="op">=</span> <span class="st">"umap"</span>, <span class="co"># could also be "tsne" or "pca"</span></span>
<span>     group.by <span class="op">=</span> <span class="st">"timepoint"</span>, <span class="co"># available column in our meta.data, from our Pluto sample data</span></span>
<span>     cols <span class="op">=</span> <span class="va">timepoint_cols</span>,</span>
<span>     pt.size <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="co"># point size for plotting</span></span>
<span>     <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>          title <span class="op">=</span> <span class="st">"Timepoint"</span>, <span class="co"># update plot title</span></span>
<span>          x <span class="op">=</span> <span class="st">"UMAP 1"</span>, <span class="co"># update x-axis label</span></span>
<span>          y <span class="op">=</span> <span class="st">"UMAP 2"</span></span>
<span>     <span class="op">)</span> <span class="co"># update y-axis label</span></span></code></pre></div>
<p><img src="https://cdn.bfldr.com/2Q1IPX6I/at/fkgzgx77673w98smxnv3vcw/scrnaseq_vignette_dimplot_group_timepoint_example" height="450"></p>
<p>Using the <code>Seurat</code> <code><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot()</a></code> function, make a
scatter plot in UMAP space grouping cells by custom annotation set
“Default cluster - res. 0.1” and splitting cells by timepoint:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">so</span>, <span class="co"># the Seurat object</span></span>
<span>     reduction <span class="op">=</span> <span class="st">"umap"</span>, <span class="co"># could also be "tsne" or "pca"</span></span>
<span>     group.by <span class="op">=</span> <span class="st">"default_clusters_res_0_1"</span>, <span class="co"># custom annotation set</span></span>
<span>     cols <span class="op">=</span> <span class="va">custom_palettes</span><span class="op">$</span><span class="va">default_clusters_res_0_1</span>, <span class="co"># custom color palette</span></span>
<span>     split.by <span class="op">=</span> <span class="st">"timepoint"</span>, <span class="co"># custom annotation set</span></span>
<span>     pt.size <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="co"># point size for plotting</span></span>
<span>     <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>          title <span class="op">=</span> <span class="st">"Default clusters - res. 0.1"</span>, <span class="co"># update plot title</span></span>
<span>          x <span class="op">=</span> <span class="st">"UMAP 1"</span>, <span class="co"># update x-axis label</span></span>
<span>          y <span class="op">=</span> <span class="st">"UMAP 2"</span></span>
<span>     <span class="op">)</span> <span class="co"># update y-axis label</span></span></code></pre></div>
<p><img src="https://cdn.bfldr.com/2Q1IPX6I/at/xwb7wrp834tm2fh9mj9s3gnp/scrnaseq_vignette_dimplot_group_split_example" height="450"></p>
</div>
<div class="section level3">
<h3 id="marker-feature-expression">Marker feature expression<a class="anchor" aria-label="anchor" href="#marker-feature-expression"></a>
</h3>
<div class="section level4">
<h4 id="violin-plot">Violin plot<a class="anchor" aria-label="anchor" href="#violin-plot"></a>
</h4>
<p>Using the <code>Seurat</code> <code><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot()</a></code> function, make a
violin plot grouping cells by custom annotation set “Default cluster -
res. 0.1”:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/VlnPlot.html" class="external-link">VlnPlot</a></span><span class="op">(</span><span class="va">so</span>, <span class="co"># the Seurat object</span></span>
<span>     features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"olig2"</span><span class="op">)</span>, <span class="co"># feature(s) of interest</span></span>
<span>     group.by <span class="op">=</span> <span class="st">"default_clusters_res_0_1"</span>, <span class="co"># custom annotation set</span></span>
<span>     cols <span class="op">=</span> <span class="va">custom_palettes</span><span class="op">$</span><span class="va">default_clusters_res_0_1</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="co"># custom color palette</span></span>
<span>     <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>          title <span class="op">=</span> <span class="st">"Expression of olig2"</span>, <span class="co"># update plot title</span></span>
<span>          x <span class="op">=</span> <span class="st">"Default clusters - res. 0.1"</span></span>
<span>     <span class="op">)</span> <span class="co"># update x-axis label</span></span></code></pre></div>
<p><img src="https://cdn.bfldr.com/2Q1IPX6I/at/b72z4b34hs68s3c7fksn9s2h/scrnaseq_vignette_vlnplot_example" height="450"></p>
</div>
<div class="section level4">
<h4 id="ridge-plot">Ridge plot<a class="anchor" aria-label="anchor" href="#ridge-plot"></a>
</h4>
<p>Using the <code>Seurat</code> <code><a href="https://satijalab.org/seurat/reference/RidgePlot.html" class="external-link">RidgePlot()</a></code> function, make
a ridge plot grouping cells by custom annotation set “Default cluster -
res. 0.1”:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/RidgePlot.html" class="external-link">RidgePlot</a></span><span class="op">(</span><span class="va">so</span>, <span class="co"># the Seurat object</span></span>
<span>     features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"olig2"</span><span class="op">)</span>, <span class="co"># feature(s) of interest</span></span>
<span>     group.by <span class="op">=</span> <span class="st">"default_clusters_res_0_1"</span>, <span class="co"># custom annotation set</span></span>
<span>     cols <span class="op">=</span> <span class="va">custom_palettes</span><span class="op">$</span><span class="va">default_clusters_res_0_1</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="co"># custom color palette</span></span>
<span>     <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>          title <span class="op">=</span> <span class="st">"Expression of olig2"</span>, <span class="co"># update plot title</span></span>
<span>          y <span class="op">=</span> <span class="st">"Default clusters - res. 0.1"</span></span>
<span>     <span class="op">)</span> <span class="op">+</span> <span class="co"># update x-axis label</span></span>
<span>     <span class="fu"><a href="https://wilkelab.org/ggridges/reference/theme_ridges.html" class="external-link">theme_ridges</a></span><span class="op">(</span>center <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="co"># center the x and y axis labels</span></span></code></pre></div>
<p><img src="https://cdn.bfldr.com/2Q1IPX6I/at/283m8qq88ns5nqf5gt5wv37x/scrnaseq_vignette_ridgeplot_example" height="450"></p>
</div>
<div class="section level4">
<h4 id="feature-plot">Feature plot<a class="anchor" aria-label="anchor" href="#feature-plot"></a>
</h4>
<p>Using the <code>Seurat</code> <code><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot()</a></code> function,
make a scatter plot in UMAP space colored by feature expression:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/FeaturePlot.html" class="external-link">FeaturePlot</a></span><span class="op">(</span><span class="va">so</span>, <span class="co"># the Seurat object</span></span>
<span>     features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"olig2"</span><span class="op">)</span>, <span class="co"># feature(s) of interest</span></span>
<span>     reduction <span class="op">=</span> <span class="st">"umap"</span>, <span class="co"># could also be "tsne" or "pca"</span></span>
<span>     pt.size <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="co"># point size for plotting</span></span>
<span>     <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>          title <span class="op">=</span> <span class="st">"Expression of olig2"</span>, <span class="co"># update plot title</span></span>
<span>          x <span class="op">=</span> <span class="st">"UMAP 1"</span>, <span class="co"># update x-axis label</span></span>
<span>          y <span class="op">=</span> <span class="st">"UMAP 2"</span></span>
<span>     <span class="op">)</span> <span class="co"># update y-axis label</span></span></code></pre></div>
<p><img src="https://cdn.bfldr.com/2Q1IPX6I/at/hksqknxt4tn3c9p4bq8x3t9/scrnaseq_vignette_featureplot_example" height="450"></p>
</div>
<div class="section level4">
<h4 id="dot-plot">Dot plot<a class="anchor" aria-label="anchor" href="#dot-plot"></a>
</h4>
<p>Using the <code>Seurat</code> <code><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot()</a></code> function, make a
dot plot colored by feature expression:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DotPlot.html" class="external-link">DotPlot</a></span><span class="op">(</span><span class="va">so</span>, <span class="co"># the Seurat object</span></span>
<span>     features <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"olig2"</span>, <span class="st">"sox19a"</span>, <span class="st">"gfap"</span>, <span class="st">"dlb"</span>, <span class="st">"nkx2.2a"</span><span class="op">)</span>, <span class="co"># feature(s) of interest</span></span>
<span>     group.by <span class="op">=</span> <span class="st">"default_clusters_res_0_1"</span></span>
<span><span class="op">)</span> <span class="op">+</span> <span class="co"># custom annotation set</span></span>
<span>     <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>          title <span class="op">=</span> <span class="st">"Key marker genes"</span>, <span class="co"># update plot title</span></span>
<span>          x <span class="op">=</span> <span class="st">"Genes"</span>, <span class="co"># update x-axis label</span></span>
<span>          y <span class="op">=</span> <span class="st">"Default clusters - res. 0.1"</span></span>
<span>     <span class="op">)</span> <span class="co"># update y-axis label</span></span></code></pre></div>
<p><img src="https://cdn.bfldr.com/2Q1IPX6I/at/x8h97bhq72mxqcqgrvbkkf5/scrnaseq_vignette_dotplot_example" height="450"></p>
<p>For more visualization tutorials using the <code>Seurat</code>
package, check out the <code>Seurat</code> visualization vignette <a href="https://satijalab.org/seurat/articles/visualization_vignette" class="external-link">here</a>.</p>
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Rani K Powers.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.2.</p>
</div>

    </footer>
</div>





  </body>
</html>
